frontend/src/components/MilestoneListItem.tsx

```diff
@@ -0,0 +1,56 @@
+import { FC, useState } from 'react'
+import styled from 'styled-components/macro'
+import { MilestoneDto } from '../dtos/MilestoneDto'
+import MilestoneEdit from './MilestoneEdit'
+
+interface MilestoneListItemProps {
+  milestone: MilestoneDto
+  fetchProject: () => void
+}
+
+const MilestoneListItem: FC<MilestoneListItemProps> = ({
+  milestone,
+  fetchProject,
+}) => {
+  const [editMode, setEditMode] = useState(false)
+
+  const switchEditMode = () => {
+    if (editMode) {
+      setEditMode(false)
+    } else setEditMode(true)
+  }
+
+  return (
+    <section>
+      {!editMode && (
+        <MilestoneListItemStyle onClick={switchEditMode}>
+          <span>{milestone.title}</span>
+          <span>{milestone.dueDate}</span>
+          <span>{milestone.dateFinished}</span>
+        </MilestoneListItemStyle>
+      )}
+      {editMode && (
+        <MilestoneEdit
+          fetchProject={fetchProject}
+          switchEditMode={switchEditMode}
+          projectTitle={milestone.projectTitle}
+          milestone={milestone}
+        />
+      )}
+    </section>
```

Comment:
```suggestion

      {!editMode && (
        <MilestoneListItemStyle onClick={switchEditMode}>
          <span>{milestone.title}</span>
          <span>{milestone.dueDate}</span>
          <span>{milestone.dateFinished}</span>
        </MilestoneListItemStyle>
      )}
      {editMode && (
        <MilestoneEdit
          fetchProject={fetchProject}
          switchEditMode={switchEditMode}
          projectTitle={milestone.projectTitle}
          milestone={milestone}
        />
      )}

```

I think you don't need the section and semantically only li are allowed as direct children to ul and ol

Reply:
The compiler does not like { javascript expresions } outside of a jsx element in  a return statment